docs/api.rst

```diff
@@ -1178,3 +1181,1395 @@ The brokers resource provides access to the current state of Kafka brokers in th
+     
+      {
+        "topics": [
+          "test1",
+	  "test2"
+        ]
+      }
+
+.. http:delete:: /consumers/(string:group_name)/instances/(string:instance)/subscription
+
+   Unsubscribe from topics currently subscribed.
+
+   Note that this request *must* be made to the specific REST proxy instance holding the consumer
+   instance.
+
+   :param string group_name: The name of the consumer group
+   :param string instance: The ID of the consumer instance
+
+   :statuscode 404:
+      * Error code 40403 -- Consumer instance not found
+
+   **Example request**:
+
+   .. sourcecode:: http
+
+      DELETE /consumers/testgroup/instances/my_consumer/subscription HTTP/1.1
+      Host: proxy-instance.kafkaproxy.example.com
+      Accept: application/vnd.kafka.v2+json, application/vnd.kafka+json, application/json
+
+   **Example response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 204 No Content
+   
+
+.. http:post:: /consumers/(string:group_name)/instances/(string:instance)/assignments
+	       
+   Manually assign a list of partitions to this consumer.
+
+   :param string group_name: The name of the consumer group
+   :param string instance: The ID of the consumer instance
+
+   :<jsonarr partitions: A list of partitions to assign to this consumer
+   :<jsonarr string partitions[i].topic: Name of the topic
+   :<jsonarr int partitions[i].partition: Partition ID
+ 
+   :statuscode 404:
+      * Error code 40403 -- Consumer instance not found
+	
+   :statuscode 409:
+      * Error code 40903 -- Subscription to topics, partitions and pattern are mutually exclusive.
+	
+   :statuscode 500:
+      * Error code 500 -- General consumer error response, caused by an exception during the
+        operation. An error message is included in the standard format which explains the cause.
+
+
+   **Example request**:
+
+   .. sourcecode:: http
+
+      POST /consumers/testgroup/instances/my_consumer/assignments HTTP/1.1
+      Host: proxy-instance.kafkaproxy.example.com
+      Content-Type: application/vnd.kafka.v2+json      
+
+      {
+        "partitions": [
+          {
+            "topic": "test",
+            "partition": 0
+          },
+          {
+            "topic": "test",
+            "partition": 1
+          }
+	  
+        ]
+      }
+      
+   **Example response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 204 No Content
+
+
+
+.. http:get:: /consumers/(string:group_name)/instances/(string:instance)/assignments
+	       
+   Get the list of partitions currently manually assigned to this consumer.
+   
+   :param string group_name: The name of the consumer group
+   :param string instance: The ID of the consumer instance
+
+   :>jsonarr partitions: A list of partitions manually to assign to this consumer
+   :>jsonarr string partitions[i].topic: Name of the topic
+   :>jsonarr int partitions[i].partition: Partition ID
+
+   :statuscode 404:
+      * Error code 40403 -- Consumer instance not found
+   :statuscode 500:
```

Comment:
Status code 500 isn't really worth repeating for every single endpoint. There's a general errors section near the top that explains general errors. And 500s generally mean you're not getting useful error info anyway, so if there isn't any info specific to this endpoint that can be given here, then it probably just adds noise.